ext/drx/drx.c

```diff
@@ -2846,23 +2898,60 @@ drx_avx2_gather_sequence_state_machine(void *drcontext,
                                        drx_state_machine_params_t *params)
 {
     switch (params->detect_state) {
-    case DRX_DETECT_RESTORE_AVX2_GATHER_EVENT_STATE_0:
+    /* First we detect the spill address for the scratch xmm reg. */
+    case DRX_DETECT_RESTORE_AVX2_GATHER_EVENT_STATE_0: {
+        params->spilled_xmm_slot_addr_reg = DR_REG_NULL;
+        params->spilled_xmm_slot_addr = NULL;
+        ptr_int_t spilled_xmm_slot_addr;
+        if (instr_is_mov_constant(&params->inst, &spilled_xmm_slot_addr) &&
+            opnd_is_reg(instr_get_dst(&params->inst, 0)) &&
+            reg_is_gpr(opnd_get_reg(instr_get_dst(&params->inst, 0)))) {
+            params->spilled_xmm_slot_addr = (void *)spilled_xmm_slot_addr;
+            params->spilled_xmm_slot_addr_reg =
+                opnd_get_reg(instr_get_dst(&params->inst, 0));
+            advance_state(DRX_DETECT_RESTORE_AVX2_GATHER_EVENT_STATE_1, params);
+        }
+        break;
+    }
+    case DRX_DETECT_RESTORE_AVX2_GATHER_EVENT_STATE_1:
+        ASSERT(params->spilled_xmm == DR_REG_NULL,
+               "Spilled xmm reg must be undetermined yet");
+        ASSERT(params->spilled_xmm_slot_addr != NULL &&
+                   params->spilled_xmm_slot_addr_reg != DR_REG_NULL,
+               "xmm spill address must be determined already");
+        if (instr_get_opcode(&params->inst) == OP_vmovdqa &&
+            opnd_is_memory_reference(instr_get_dst(&params->inst, 0)) &&
+            opnd_get_base(instr_get_dst(&params->inst, 0)) ==
```

Comment:
opnd_is_memory_reference() returns true for opnd types that don't have bases: safer to change it to opnd_is_base_disp()

Reply:
Done.